tests/main/snap-userd-socket-activation/task.yaml

```
@@ -0,0 +1,47 @@
+summary: Check that root can activate userd via socket activation
+
+systems:
+    - -ubuntu-core-*
+    - -ubuntu-14.04-*
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    mkdir -p "$USER_RUNTIME_DIR"
+    chmod u=rwX,go= "$USER_RUNTIME_DIR"
+    chown test:test "$USER_RUNTIME_DIR"
+    systemctl start "user@${TEST_UID}.service"
+
+restore: |
+    systemctl stop "user@${TEST_UID}.service"
+    rm -rf "${USER_RUNTIME_DIR:?}"/* "${USER_RUNTIME_DIR:?}"/.[!.]*
+
+execute: |
+    systemctl_user() {
+        su -l -c "XDG_RUNTIME_DIR=\"${USER_RUNTIME_DIR}\" systemctl --user $*" test
+    }
+    echo "Initially snap userd is not running"
+    ! systemctl_user is-active snap-userd.service
```

Please use `not`: see https://github.com/snapcore/snapd/pull/6908 and https://github.com/snapcore/snapd/pull/6926 for context 

```suggestion
    not systemctl_user is-active snap-userd.service
```

---

So, I made this change but it actually causes the check to malfunction.  As "systemctl_user" is a shell function, it wasn't available to the "not" shell script.  The test continued to pass since the "command not found" error was inverted to a success.

I guess changing this to `if ! command; then exit 1; fi` or similar would work.